app/src/test/java/com/android/sample/model/user/UserViewModelTest.kt

```diff
@@ -561,4 +568,279 @@ class UserViewModelTest {
+            images =
+                mutableMapOf(
+                    PRODUCT_FRONT_IMAGE_NORMAL_URL to "https://display_normal",
+                    PRODUCT_FRONT_IMAGE_THUMBNAIL_URL to "https://display_thumbnail",
+                    PRODUCT_FRONT_IMAGE_SMALL_URL to "https://display_small"))
+
+    // Add the ingredient to the ingredient list
+    userViewModel.addIngredient(ingredientFalse)
+    userViewModel.addIngredient(ingredientCorrect)
+
+    // Verify that the ingredient list contains the ingredient with the updated quantity
+    val updatedIngredient =
+        userViewModel.ingredientList.value.find { it.first.barCode == ingredientCorrect.barCode }
+    assertNotNull(updatedIngredient)
+    TestCase.assertEquals("100g", updatedIngredient?.second)
+  }
+
+  @Test
+  fun addACorrectIngredientAndAFalseIngredientTest() {
+    // Create an initial ingredient
+    val ingredientCorrect =
+        Ingredient(
+            barCode = 123456L,
+            name = "Test Ingredient",
+            brands = "Brand",
+            quantity = "100g",
+            categories = listOf("Category1"),
+            images =
+                mutableMapOf(
+                    PRODUCT_FRONT_IMAGE_NORMAL_URL to "https://display_normal",
+                    PRODUCT_FRONT_IMAGE_THUMBNAIL_URL to "https://display_thumbnail",
+                    PRODUCT_FRONT_IMAGE_SMALL_URL to "https://display_small"))
+    val ingredientFalse =
+        Ingredient(
+            barCode = 123456L,
+            name = "Test Ingredient",
+            brands = "g",
+            categories = listOf("Category1"),
+            images =
+                mutableMapOf(
+                    PRODUCT_FRONT_IMAGE_NORMAL_URL to "https://display_normal",
+                    PRODUCT_FRONT_IMAGE_THUMBNAIL_URL to "https://display_thumbnail",
+                    PRODUCT_FRONT_IMAGE_SMALL_URL to "https://display_small"))
+
+    // Add the ingredient to the ingredient list
+    userViewModel.addIngredient(ingredientCorrect)
+    userViewModel.addIngredient(ingredientFalse)
+
+    // Verify that the ingredient list contains the ingredient with the updated quantity
+    val updatedIngredient =
+        userViewModel.ingredientList.value.find { it.first.barCode == ingredientCorrect.barCode }
+    assertNotNull(updatedIngredient)
+    TestCase.assertEquals("100g", updatedIngredient?.second)
+  }
+
+  @Test
+  fun addNullRegexIngredientTest() {
+    // Create an initial ingredient
+    val ingredient =
+        Ingredient(
+            barCode = 123456L,
+            name = "Test Ingredient",
+            brands = "Brand",
+            quantity = "s",
+            categories = listOf("Category1"),
+            images =
+                mutableMapOf(
+                    PRODUCT_FRONT_IMAGE_NORMAL_URL to "https://display_normal",
+                    PRODUCT_FRONT_IMAGE_THUMBNAIL_URL to "https://display_thumbnail",
+                    PRODUCT_FRONT_IMAGE_SMALL_URL to "https://display_small"))
+
```

Comment:
I've updated the tests with the sample ingredients, but I couldn't do the same for the Regex cases since the ingredient is used only once. Is that okay with you?

Reply:
That's fine with me